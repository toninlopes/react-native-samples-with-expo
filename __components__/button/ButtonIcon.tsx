import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import type { ButtonIconProps } from './types';

const ButtonIcon: React.FC<ButtonIconProps> = ({ name }) => (
  <View style={styles.view}>
    <Ionicons name={name} size={28}/>
  </View>
);

const styles = StyleSheet.create({
  view: {
    paddingHorizontal: 12,
  },
});

export default ButtonIcon;