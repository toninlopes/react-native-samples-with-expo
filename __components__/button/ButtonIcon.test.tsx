import React from 'react';
import renderer from 'react-test-renderer';

import ButtonIcon from './ButtonIcon';

describe('ButtonIcon', () => {

  test('When there is default name', () => {
    const button = renderer.create(<ButtonIcon name='timer-outline' />).toJSON();
    expect(button).toMatchSnapshot();
  });
});