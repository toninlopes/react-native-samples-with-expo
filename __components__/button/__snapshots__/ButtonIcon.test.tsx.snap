// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon When there is default name 1`] = `
<View
  style={
    {
      "paddingHorizontal": 12,
    }
  }
>
  <Text />
</View>
`;
